### PR TITLE
Upgrade to Flyway 7.9.0

### PR DESCRIPTION
**Pull request by gsmet**

### CODE REVIEW

File: extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/QuarkusPathLocationScanner.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public QuarkusPathLocationScanner(Collection<Location> locations) {
                 scannedResources.add(new ClassPathResource(null, migrationFile, classLoader, StandardCharsets.UTF_8));
             } else if (migrationFile.startsWith(Location.FILESYSTEM_PREFIX)) {
                 if (fileSystemScanner == null) {
-                    fileSystemScanner = new FileSystemScanner(StandardCharsets.UTF_8, false);
+                    fileSystemScanner = new FileSystemScanner(StandardCharsets.UTF_8, false, false, false);
```
Comment (gsmet) [inline extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/QuarkusPathLocationScanner.java:45]: We will probably want to add configuration properties for these but I'll let our community add them if they need them.

Comment (geoand) [reply extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/QuarkusPathLocationScanner.java:45]: 👍🏼

**Review by geoand — APPROVED** [review]
Assuming CI is happy, let's go with it!

**Comment by gsmet** [comment]
@famod something is weird with GIB. The build here was successful but... the Data3 native tests build was not executed.

And now we have a failure here: https://github.com/quarkusio/quarkus/pull/17297 .

When we had a BOM change, weren't we supposed to run a full build?

**Comment by gsmet** [comment]
(And even not considering the BOM change, given the Flyway extension changed, the Data3 tests should have been run)

**Comment by famod** [comment]
I'll try to have a look tomorrow.

**Comment by zakkak** [comment]
FTR this PR seems to break the following native tests:

* Quartz
* Flyway
* Hibernate - Multitenancy

**Comment by geoand** [comment]
We should probably revert for the time being

**Comment by zakkak** [comment]
> We should probably revert for the time being

The fix looks like as simple as:
```diff
diff --git a/extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/graal/ScannerSubstitutions.java b/extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/graal/ScannerSubstitutions.java
index 1840d08819..dc184d56af 100644
--- a/extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/graal/ScannerSubstitutions.java
+++ b/extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/graal/ScannerSubstitutions.java
@@ -19,8 +19,10 @@ public final class ScannerSubstitutions {
     @Substitute
     public ScannerSubstitutions(Class<?> implementedInterface, Collection<Location> locations, ClassLoader classLoader,
             Charset encoding,
+            boolean detectEncoding,
             boolean stream,
-            ResourceNameCache resourceNameCache, LocationScannerCache locationScannerCache) {
+            ResourceNameCache resourceNameCache, LocationScannerCache locationScannerCache,
+            boolean throwOnMissingLocations) {
         throw new IllegalStateException("'org.flywaydb.core.internal.scanner.Scanner' is never used in Quarkus");
     }
 }

```

I am going to open a PR once I test it.

**Comment by famod** [comment]
Ok, there is something wrong in the `Calculate matrix from native-tests.json` step: https://github.com/quarkusio/quarkus/runs/2600605046?check_suite_focus=true

error

I'll need to check what's that about. Pretty dodgy that it doesn't fail properly...

**Comment by famod** [comment]
Good news: Reproducible locally. Looking at it now.

**Comment by famod** [comment]
#17319 should fix this ⬆️ (tested locally with multiple inputs, also with the long list from this PR).

**Pull request merged**